### PR TITLE
Various refactoring of code following OAuth1 implementation

### PR DESCRIPTION
**Pull request by AndyButland**
This PR is an update to #28, providing some refactoring and fixes found in review (the specifics are detailed in the commit messages).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,11 +45,15 @@ Install-Package Umbraco.AuthorizedServices
...
 ```
 
-
 ### App Creation
 
 Services that this package are intended to support will offer an OAuth authentication and authorization flow against an "app" that the developer will need to create with the service.  From this various information will be available, including for example a "client ID" and "client secret" that will need to be applied in configuration.
 
+When creating the app it will usually be necessary to configure a call back URL. You should use the following:
+
+- For OAuth2: `/api/AuthorizedServiceResponse/HandleOAuth2IdentityResponse`
```
Comment (acoumb) [inline README.md:54]: I think this needs to be `/umbraco/api/AuthorizedServiceResponse/HandleOAuth2IdentityResponse` and `/umbraco/api/AuthorizedServiceResponse/HandleOAuth1IdentityResponse`

**Review by acoumb — APPROVED** [review]
Looks good, just a small comment on the URLs in the `readme.md`

File: README.md (hunk 1 above)
Comment (AndyButland) [reply README.md:54]: Agreed - please can you change that and then merge if you are happy with the other changes/have confirmed I didn't break anything.

Comment (acoumb) [reply README.md:54]: Sure thing, I will merge them in after I run some tests with Flickr. Thank you.

**Pull request merged**